src/Http/Http/src/DefaultHttpContext.cs

```diff
@@ -90,7 +90,7 @@ public void Uninitialize()
         private IHttpRequestIdentifierFeature RequestIdentifierFeature =>
             _features.Fetch(ref _features.Cache.RequestIdentifier, _newHttpRequestIdentifierFeature);
 
-        public override IFeatureCollection Features => _features.Collection;
+        public override IFeatureCollection Features => _features.Collection ?? ContextDisposed();
```

Comment:
Similarly, why not use a single body here instead of two methods?
```suggestion
        public override IFeatureCollection Features
        {
            get
            {
                if (_features.Collection == null)
                {
                    throw new ObjectDisposedException(nameof(HttpContext), $"Request has finished and {nameof(HttpContext)} disposed.");
                }

                return _features.Collection;
            }
        }
```

Reply:
The Jit has special optimizations for a void method that only throws.

1. It recognises it as cold code (shouldn't be throwing exceptions in common paths), so generates the branches for it to be unpredicted (also does this with regular exception throws)
2. As it recognised it as cold code it won't inline it; whereas regular throws are not split out (Jit doesn't split methods); so the instruction cache is hotter (exception throws are quite big).
3. It doesn't generate the asm for it until the exception is thrown (faster Jitting).
4. As it recognises it as non-returning, it doesn't bother trying to sort out the stack after the call (popping regsiters back), meaning the code is hotter.
5. If the method is generic, moving the throws into a non-returning method in a non-generic class drastically reduces the asm size generated (doesn't do it for every type).
6. If the method inlines, moving the the throws into a non-returning method reduces the asm size generated as it doesn't regenerate it for every caller.

tl;dr makes the CPU instruction cache better; generics smaller; inlines smaller; faster Jitting